hudi-hadoop-common/src/test/java/org/apache/hudi/metadata/TestHoodieTableMetadataUtil.java

```diff
@@ -652,6 +652,27 @@ public void testValidateDataTypeForPartitionStats() {
         .name("dateField").type(dateFieldSchema).noDefault()
         .endRecord();
     assertTrue(HoodieTableMetadataUtil.isColumnTypeSupported(schema.getField("dateField").schema(), Option.empty()));
+
+    // Test for logical decimal type with allowed precision and scale
+    schema = Schema.create(Schema.Type.BYTES);
+    LogicalTypes.Decimal decimalType = LogicalTypes.decimal(30, 15);
+    decimalType.addToSchema(schema);
+    // Expect the column to be supported.
+    assertTrue(HoodieTableMetadataUtil.isColumnTypeSupported(schema, Option.of(HoodieRecord.HoodieRecordType.AVRO)));
+
+    // Test for logical decimal type with precision and scale exceeding the limit
```

Comment:
We should have tests where just 1 or the other exceeds as well to ensure the scale condition is hit in our testing.

Reply:
can you TAL. I have added them.